src/core/ddsc/include/dds/ddsc/dds_loan_api.h

```diff
@@ -0,0 +1,54 @@
+
+// API extension
+// defines functions needed for loaning and shared memory usage
+
+#ifndef _DDS_LOAN_API_H_
+#define _DDS_LOAN_API_H_
+
+#include "dds/ddsc/dds_public_types.h"
+#include "dds/ddsrt/retcode.h"
+#include "dds/export.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+// MAKI: declare prototype in data_allocator with deprecated comment
+
+/** @brief Check if a Loan is available to reader/writer
+ * The loan is available if the shared memory is enabled and all the constraints
+ * to enable shared memory are met and the type is fixed
+ *
+ * @param[in] entity the handle of the entity
+ *
+ * @returns loan available or not
+ */
+DDS_EXPORT bool dds_is_loan_available(const dds_entity_t entity);
+
+/** @brief Check if a shared memory is available to reader/writer.
+ *
+ * @note shared memory available and loan available have different meaning
+ * 1)We may not be able to use the loan API due to type constraints (not fixed)
+ * 2) We may also allow using the loan API but without shared memory
```

Comment:
The note is confusing to me, this should say something in the lines that `dds_loan_sample` API is allowed to be used only when `dds_is_loan_available`, correct? or am I missing something here?

Reply:
Clarified notes.